src/components/settings/ProposersList/index.tsx

```diff
@@ -51,7 +53,9 @@ const ProposersList = () => {
                         onClick={() => onDelete(proposer.delegate)}
                         color="error"
                         size="small"
-                        disabled={!isOk}
+                        disabled={
+                          !isOk || (wallet?.address !== proposer.delegate && wallet?.address !== proposer.delegator)
```

Comment:
```suggestion
                          !isOk || wallet?.address !== proposer.delegator
```

If I understand correctly, we could rewrite like this if we add `allowProposer={false}` to `CheckWallet`? Or can proposers remove themselves?

Reply:
Yes, the API allows proposers to remove themselves so I left that possiblity in the UI too